### DISCUSSION TITLE
Copilot activation failed

### DISCUSSION THREAD

**Discussion by ar0matix** [community/community#18223] [Copilot Conversations]
I was trying to test copilot for JS but it says that "Copilot activation failed", when I click on the icon it says "Please upgrade your Copilot extension to continue using this service." but even whether when I did update that, I'm still getting the error. Plz help.

**Comment by DavidYang2149** [comment]
Hello.
What is your IDE tool?
if you use VSCode, Menu -> Help -> type 'Check for Updates' and update your VSCode version.

**Reply by PaulDotSH** [reply]
You are probably using an outdated version of vscode or a FOSS fork of it like vscodium. Also depends on your distro if you are using linux, kali linux's vscodium is outdated ATM. You can install the proprietary version of vscode and it should work.

**Comment by MrKrishnaAgarwal** [comment]
Hey,
You can try signing up here: https://github.com/features/copilot/signup

**Comment by ar0matix** [comment]
Hey guys, I solved it, the issue was that I had to update my vscode and copilot. Btw, thanks for your time.

**Reply by MrKrishnaAgarwal** [reply]
Great

**Reply by DavidYang2149** [reply]
Good for you! 😄

**Reply by ChrisPorter87** [reply]
@ar0matix  I'm having that same issue.  When I try to search for updates for VS code, I get this Cannot update while running on a read-only volume. The application is on a read-only volume. Please move the application and try again. If you're on macOS Sierra or later, you'll need to move the application out of the Downloads directory. This might mean the application was put on quarantine by macOS. See [this link](https://github.com/microsoft/vscode/issues/7426#issuecomment-425093469) for more information.  Did you see that too and if so how did you solve it?

**Reply by ar0matix** [reply]
@ChrisPorter87 I'm sorry buddy I use windows

**Reply by PythonCancer** [reply]
My IDE is vscode and the operating system is win11. After I reinstall the system, copilot cannot be activated. The error is: copilot activation failed. How can I solve this problem?

**Reply by ar0matix** [reply]
> My IDE is vscode and the operating system is win11. After I reinstall the system, copilot cannot be activated. The error is: copilot activation failed. How can I solve this problem?

Did u try updating your copilot? And also your vscode?

**Comment by GeneralChauhan** [comment]
You should check for updates
